### PR TITLE
Exploring node communication and async streams

### PR DESCRIPTION
**Pull request by prince-chrismc**
This was my POC getting two node to talk. The behavior no longer matches the demo but every is still working.

Under the intent of #57

### CODE REVIEW

**Comment by prince-chrismc** [comment]
@efrisch does this make any sense to merge with your work on #169 ?

**Comment by prince-chrismc** [comment]
> @efrisch does this make any sense to merge with your work on #169 ?

**Review by efrisch — APPROVED** [review]
👍

File: pyrsia_node/readme.md [hunk 1]
```diff
@@ -12,13 +12,26 @@ The daemon running everything.
...
+   - tag it to prepare for push to pyrsia node: `docker tag alpine localhost:7878/alpine`
+   - push it to pyrsia node: `docker push localhost:7878/alpine`
+   - remove all local alpine images: `docker rmi alpine and docker rmi localhost:7878/alpine`
+   - pull the image again, this time from pyrsia node: `docker pull localhost:7878/alpine`
+   - verify it works: `docker run -it localhost:7878/alpine cat /etc/issue`
+
+### Manually interacting with Docker API
+
+1. Open a terminal and start a pyrsia node with: `RUST_LOG=pyrsia cargo run -q`
+2. Start 2 more nodes with different ports
```
Comment (MitaliBo) [inline pyrsia_node/readme.md:29]: here, its good to mention the command line argument to start new ports or manually dialing a peer

Comment (prince-chrismc) [reply pyrsia_node/readme.md:29]:
```suggestion
2. Start 2 more nodes with different ports by adding `-p ####` to the command above
```

File: pyrsia_node/readme.md [hunk 2]
```diff
@@ -12,13 +12,26 @@ The daemon running everything.
...
+   - tag it to prepare for push to pyrsia node: `docker tag alpine localhost:7878/alpine`
+   - push it to pyrsia node: `docker push localhost:7878/alpine`
+   - remove all local alpine images: `docker rmi alpine and docker rmi localhost:7878/alpine`
+   - pull the image again, this time from pyrsia node: `docker pull localhost:7878/alpine`
+   - verify it works: `docker run -it localhost:7878/alpine cat /etc/issue`
+
+### Manually interacting with Docker API
+
+1. Open a terminal and start a pyrsia node with: `RUST_LOG=pyrsia cargo run -q`
+2. Start 2 more nodes with different ports by adding `-p ####` to the command above
```
Comment (prince-chrismc) [inline pyrsia_node/readme.md:29]:
```suggestion
2. Start 3 more nodes with different ports by adding `-p ####` to the command above
```

**Comment by prince-chrismc** [comment]
Update every, should be ready for final review 🙏

**Review by fishseabowl — APPROVED** [review]
It looks good for me

File: pyrsia_node/src/main.rs [hunk 3]
```diff
@@ -173,10 +176,16 @@ async fn main() {
             tokio::select! {
                 line = stdin.next_line() => Some(EventType::Input(line.expect("can get line").expect("can read line from stdin"))),
                 message = rx.recv() => Some(EventType::Message(message.expect("message exists"))),
-               // response = rx.recv() => Some(EventType::Response(response.expect("response exists"))),
+
+                new_hash = blobs_need_hash.select_next_some() => {
+                    debug!("Looking for {}", new_hash);
+                    swarm.behaviour_mut().lookup_blob(new_hash).await;
+                    None
+                },
```
Comment (mgrand) [inline pyrsia_node/src/main.rs:184]: New things in this main methods really should start going into separate functions.  It is really long.

**Review by mgrand — APPROVED** [review]

**Pull request merged**